index.html

```diff
@@ -654,5 +654,400 @@ <h2>Exposing change of MediaStreamTrack configuration</h2>
+          </dd>
+        </dl>
+      </section>
+    </section>
+    <section>
+      <h3>{{HumanFaceDetectionMode}}</h3>
+<pre class="idl">
+enum HumanFaceDetectionMode {
+  "none",
+  "bounding-box",
+};</pre>
+      <section class="notoc">
+        <h4>{{HumanFaceDetectionMode}} Enumeration Description</h4>
+        <dl data-dfn-for="HumanFaceDetectionMode" data-link-for="HumanFaceDetectionMode">
+          <dt><dfn><code>none</code></dfn></dt>
+          <dd>
+            <p>This source does not offer object detection for human faces.
+            For setting, this is interpreted as a command to turn off
+            the setting of human face detection metadata.</p>
+          </dd>
+          <dt><dfn><code>bounding-box</code></dfn></dt>
+          <dd>
+            <p>This source offers detection of the bounding box of human faces or such a mode is requested.</p>
+          </dd>
+        </dl>
+      </section>
+    </section>
+    <section>
+      <h3>{{HumanFaceLandmarkDetectionMode}}</h3>
+<pre class="idl">
+enum HumanFaceLandmarkDetectionMode {
+  "none",
+  "center-point",
+};</pre>
+      <section class="notoc">
+        <h4>{{HumanFaceLandmarkDetectionMode}} Enumeration Description</h4>
+        <dl data-dfn-for="HumanFaceLandmarkDetectionMode" data-link-for="HumanFaceLandmarkDetectionMode">
+          <dt><dfn><code>none</code></dfn></dt>
+          <dd>
+            <p>This source does not offer object detection for human face landmarks.
+            For setting, this is interpreted as a command to turn off
+            the setting of human face landmark detection metadata.</p>
+          </dd>
+          <dt><dfn><code>center-point</code></dfn></dt>
+          <dd>
+            <p>This source offers detection of the center point of human face landmarks, or such a mode is requested.</p>
+          </dd>
+        </dl>
+      </section>
+    </section>
+    <section>
+      <h3>Constrainable Properties</h3>
+      <ol>
+        <li>
+          <p><dfn>Face detection mode</dfn> describes which face properties
+          are to be detected and set in the metadata for the video frame.</p>
+        </li>
+        <li>
+          <p><dfn>Face landmark detection mode</dfn> describes which face landmark
+          properties are to be detected and set in the metadata for the video frame.</p>
+        </li>
+      </ol>
+    </section>
+    <section>
+      <h3>Examples</h3>
+      <pre class="example">
+// main.js:
+// Check if face detection is supported by the browser
+const supports = navigator.mediaDevices.getSupportedConstraints();
+if (supports.humanFaceDetectionMode) {
+  // Browser supports face contour detection.
+} else {
+  throw('Face contour detection is not supported');
+}
+
+// Open camera with face detection enabled
+const stream = await navigator.mediaDevices.getUserMedia({
+  video: { humanFaceDetectionMode: 'bounding-box' }
+});
+const [videoTrack] = stream.getVideoTracks();
+
```

Comment:
I care a lot about examples, so apologies for all the comments.

I think there's a false assumption here that since `supports.humanFaceDetectionMode` is `true` here, then `videoTrack.getSettings().humanFaceDetectionMode` MUST be `'bounding-box'`.

But I don't think that's right. That would imply that the face tracking feature is implemented entirely in the browser for all camera devices without any reliance on specific device or driver hardware. Even if this is true for some user agents, I see no reason to bake this assumption into the API, when the `supports` mechanism was never meant to be where features were negotiated.

It also doesn't allow for user agents to support this for _some_ special hardware cameras but not others, nor does it appear to allow user agents to allow users to turn off these features. We should also discourage "must have" code patterns IMHO. Most likely, if face tracking cannot be gotten, then most camera apps should proceed anyway.

I think instead the model we want to promote is to apply the constraints and then read back the result with `getSettings` and react to that if necessary if so desired. e.g.
```js
if (videoTrack.getSettings().humanFaceDetectionMode != 'bounding-box') {
  throw('Face detection is not supported');
}
```
I'd prefer this added to the examples instead of using getSupportedConstraints. This also simplifies subsequent code, as I show below.

Reply:
I completely agree, and appreciate your detailed feedback. Fixing the examples.